### PR TITLE
config fluentd for multi-cluster

### PR DESCRIPTION
**Pull request by shihchang**
# Description
* multi-cluster - application-operator/mcagent update system fluentd in managed cluster with mc-registration-secret
* config fluentd in admin cluster to use oidc-proxy
* rename index pattern to verrazzano-*
* helido and containerizedWorkload/springboot functional test
* fix test with date suffix indices

Fixes VZ-2394

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/mcagent/mcagent.go [hunk 1]
```diff
@@ -261,6 +262,147 @@ func (s *Syncer) updateDeployment(name string) {
...
+		}
+	} else {
+		secretVersion = regSecret.ResourceVersion
+	}
+	secretVersionEnv := getContainerEnvValue(&daemonSet.Spec.Template.Spec.Containers, registrationSecretVersion)
+	// CreateOrUpdate updates the deployment if cluster name or es secret version changed
+	if secretVersionEnv != secretVersion {
+		controllerutil.CreateOrUpdate(s.Context, s.LocalClient, &daemonSet, func() error {
+			s.Log.Info(fmt.Sprintf("Update the DaemonSet %s, registration secret version from %q to %q", loggingName, secretVersionEnv, secretVersion))
+			daemonSet = *updateLoggingDaemonSet(constants.MCRegistrationSecret, secretVersion, &daemonSet)
```
[the remark this answers is not in the document]
Comment (shihchang) [reply application-operator/mcagent/mcagent.go:289]: What does un-registration means? does the cluster become admin or remain managed?

File: application-operator/mcagent/mcagent.go [hunk 2]
```diff
@@ -261,6 +262,147 @@ func (s *Syncer) updateDeployment(name string) {
...
+		}
+	}
+	return ds
+}
+
+func updateEnv(newSecret string, old []corev1.EnvVar) []corev1.EnvVar {
+	var new []corev1.EnvVar
+	for _, env := range old {
+		if env.Name == "CLUSTER_NAME" {
+			new = append(new, corev1.EnvVar{
```
Comment (guoyongzhang) [inline application-operator/mcagent/mcagent.go:322]: maybe back to local, after unregister?

Comment (guoyongzhang) [reply application-operator/mcagent/mcagent.go:322]: similar for other env for unregister, go back to original helm config?

Comment (shihchang) [reply application-operator/mcagent/mcagent.go]: back to local

File: application-operator/mcagent/mcagent.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shihchang) [reply application-operator/mcagent/mcagent.go:289]: change all the env back to default

File: application-operator/mcagent/mcagent.go [hunk 3]
```diff
@@ -261,6 +262,172 @@ func (s *Syncer) updateDeployment(name string) {
...
+
+func updateVolumes(newSecret string, old []corev1.Volume) []corev1.Volume {
+	var new []corev1.Volume
+	for _, vol := range old {
+		if vol.Name == "secret-volume" {
+			new = append(new, corev1.Volume{
+				Name: vol.Name,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: newSecret},
```
Comment (guoyongzhang) [inline application-operator/mcagent/mcagent.go:421]: would secretName always be MCRegistrationSecret?

File: application-operator/mcagent/mcagent.go [hunk 4]
```diff
@@ -261,6 +262,172 @@ func (s *Syncer) updateDeployment(name string) {
...
+	if secretVersionEnv != secretVersion {
+		controllerutil.CreateOrUpdate(s.Context, s.LocalClient, &daemonSet, func() error {
+			s.Log.Info(fmt.Sprintf("Update the DaemonSet %s, registration secret version from %q to %q", loggingName, secretVersionEnv, secretVersion))
+			daemonSet = *updateLoggingDaemonSet(constants.MCRegistrationSecret, secretVersion, &daemonSet)
+			return nil
+		})
+	}
+}
+
+func getContainerEnvValue(containers *[]corev1.Container, envName string) string {
```
Comment (guoyongzhang) [inline application-operator/mcagent/mcagent.go:295]: we should use either getContainerEnvValue or getEnvValue to be consistent.  we don't have both around.

File: application-operator/mcagent/mcagent_test.go [hunk 5]
```diff
@@ -452,3 +452,131 @@ func expectGetAPIServerURLCalled(mock *mocks.MockClient) {
 			return nil
 		})
 }
+
```
Comment (guoyongzhang) [inline application-operator/mcagent/mcagent_test.go:455]: updateEnv and updateVolumes are complex.  should there be test cases for them?  either inside TestSyncer_configureLogging or by themselves.

File: application-operator/mcagent/mcagent.go (hunk 3 above)
Comment (shihchang) [reply application-operator/mcagent/mcagent.go]: fixed and tested

File: application-operator/mcagent/mcagent.go (hunk 4 above)
Comment (shihchang) [reply application-operator/mcagent/mcagent.go]: renamed getContainerEnvValue to getEnvValue replacing all existing getEnvValue

File: application-operator/mcagent/mcagent_test.go (hunk 5 above)
Comment (shihchang) [reply application-operator/mcagent/mcagent_test.go:456]: test added

**Review by guoyongzhang — APPROVED** [review]

**Pull request merged**